queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/shares.py

```diff
@@ -71,6 +71,79 @@ def update_share_structure(business: Business, share_structure: Dict) -> Optiona
     return err
 
 
+def update_share_structure_correction(business: Business, share_structure: Dict) -> Optional[List]:
+    """Manage the share structure for a business.
+
+    Assumption: The structure has already been validated, upon submission.
+
+    Other errors are recorded and will be managed out of band.
+    """
+    if not business or not share_structure:
+        # if nothing is passed in, we don't care and it's not an error
+        return None
+
+    err = []
+
+    inclusion_entries = []
+    exclusion_entries = []
+    # Delete the ones that are present in db but not in the json and create the ones in json but not in db.
+    if resolution_dates := share_structure.get('resolutionDates'):
+        # Two lists of dates in datetime format
+        business_dates = [item.resolution_date for item in business.resolutions]
+        parsed_dates = [parse(resolution_dt).date() for resolution_dt in resolution_dates]
+
+        # Dates in both db and json
+        inclusion_entries = [business.resolutions[index] for index, date in enumerate(business_dates)
+                             if date in parsed_dates]
+        if len(inclusion_entries) > 0:
+            business.resolutions = inclusion_entries
+
+        # Dates in json and not in db
+        exclusion_entries = [date for date in parsed_dates if date not in business_dates]
+
+        resolution_dates = exclusion_entries
+
+        for resolution_dt in resolution_dates:
+            try:
+                d = Resolution(
+                    resolution_date=resolution_dt,
+                    resolution_type=Resolution.ResolutionType.SPECIAL.value
+                )
+                business.resolutions.append(d)
+            except (ValueError, OverflowError):
+                err.append(
+                    {'error_code': 'FILER_INVALID_RESOLUTION_DATE',
+                     'error_message': f"Filer: invalid resolution date:'{resolution_dt}'"}
+                )
+
+    if share_classes := share_structure.get('shareClasses'):
+        # business share classes array of jsons
+        business_share_ids = [share_class.id for share_class in business.share_classes]
+        share_structure_ids = [share_class.get('id') for share_class in share_classes]
+
+        # Dates in both db and json
+        inclusion_entries = [share_class for share_class in business.share_classes
+                             if share_class.id in share_structure_ids]
+        if len(inclusion_entries) > 0:
+            business.share_classes = inclusion_entries
```

Comment:
Will this work if there is a new share series under an existing share class json?